### DISCUSSION TITLE
KeyList from Keypad Touch

### DISCUSSION THREAD

**Discussion by Master-Player** [hsakoh/switchbot-mqtt#35] [Q&A] [answered]
Hello,
is it possible to display the device attributes, especially the KeyList, in Home Assistant on the Keypad Touch? Then it would be possible to delete the correct ID for passcodes again via HA if necessary.

Thank You

**Comment by hsakoh** [comment, marked as answer]
I have just released v1.0.21.

Improvements to Keypad and Keypad Touch.

* Changed the method of specifying the Key ID in the DeleteKey command from Mqtt.Number to Mqtt.Select.
   * After adding or deleting keys, the dropdown list options can be updated by pressing the "Reload Keys" button.
      * Note: The options are not updated automatically. This is because enumerating the keys requires calling the ["Get device list" API](https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#keypad), which is expected to incur a high cost on the SwitchBot side.
   ![image](https://github.com/user-attachments/assets/e8f56f48-001c-4baa-9aa1-6c80f3200687)

   * To apply the input format change, you need to delete the corresponding command device in the MQTT integration and restart the addon (or service).<br>![image](https://github.com/user-attachments/assets/b6376ca5-b012-4d29-a0a8-e1b66dece463)

* Added a screen for viewing, adding, and deleting passcodes on the Ingress page.
   * It also supports passcode decryption.
![image](https://github.com/user-attachments/assets/c1ae4a61-fc38-4f65-84c0-a46f243768de)
![image](https://github.com/user-attachments/assets/7f5d88bb-afe7-46e4-bb83-a2d74d67a656)

**Comment by Master-Player** [comment]
works like a charm, thank you
